lib/rrdd_plugin.ml

```diff
@@ -16,9 +16,48 @@ open Xapi_stdext_threads.Threadext
 
 open Xapi_stdext_unix
 
-(* This exception is setup to be raised on sigint by Process.initialise, and is
-   used to cancel the synchronous function Reporter.start. *)
+(* This exception is setup to be raised on sigint by Process.initialise,
+ * and is used to cancel the synchronous function Reporter.start. *)
 exception Killed
+let killed = ref false (* CA-309024, Killed might escape *)
+
+let signal_name signum =
+  let signals =
+    let t = Hashtbl.create 30 in
+    let map =
+      [ ("SIGABRT", -1)
```

Comment:
Good idea with mapping these to strings, I always get confused which negative signal number maps to what name.
This could use the constants from `Sys`, e.g. `Sys.sigabrt` instead of the numbers though.

Reply:
I've added that.